src/migrations.js

```diff
@@ -0,0 +1,31 @@
+import { produce, setAutoFreeze } from 'immer';
+
+export const migrate = (
+  data,
+  migrations,
+) => {
+  setAutoFreeze(false);
+
+  let version = data._migrationVersion || -1;
+  const toVersion = migrations.migrationVersion
+
+  if (typeof toVersion !== 'number') {
+    throw new Error('No migration version found');
+  }
+
+  while (version < toVersion) {
+    const nextVersion = version + 1;
+    const migrator = migrations[nextVersion];
+
+    if (!migrator) {
+      throw new Error(`No migrator found for \`migrationVersion\` ${nextVersion}`);
+    }
+
+    data = produce(data, migrator);
+    data._migrationVersion = nextVersion;
+    version = data._migrationVersion;
+  }
+
+  setAutoFreeze(true);
```

Comment:
Turn immer's object freezing capabilities back on now that we've set the latest `migrationVersion` and are done migrating